### PR TITLE
Cleaned up tracking of deleted node/rel ids in tx state

### PR DESCRIPTION
**Pull request by lutovich**
This is a minor clean up commit that:
- removes not relevant comments
- changes sets of ids to primitive sets
- adds couple tests

### CODE REVIEW

File: community/cypher/cypher/src/test/scala/org/neo4j/cypher/MutatingIntegrationTest.scala [hunk 1]
```diff
@@ -478,4 +478,14 @@ return distinct center""")
 
     assertStats(result, nodesCreated = 1)
   }
+
+  test("should be possible to remove nodes created in the same query") {
+    val result = execute(
+      """CREATE (a)-[:FOO]->(b)
+         WITH a, b
```
Comment (davidegrohmann) [inline community/cypher/cypher/src/test/scala/org/neo4j/cypher/MutatingIntegrationTest.scala]: `WITH a, b`?

Comment (lutovich) [inline community/cypher/cypher/src/test/scala/org/neo4j/cypher/MutatingIntegrationTest.scala]: Yes. It does not make any semantic difference but syntax requires `WITH` between `CREATE` and `MATCH`

Comment (davidegrohmann) [inline community/cypher/cypher/src/test/scala/org/neo4j/cypher/MutatingIntegrationTest.scala]: `WITH *`perhaps?

Comment (lutovich) [inline community/cypher/cypher/src/test/scala/org/neo4j/cypher/MutatingIntegrationTest.scala]: Sure

**Pull request merged**